pep-0501.txt

```diff
@@ -383,27 +457,87 @@ errors all raise SyntaxError.
 
 Unmatched braces::
 
-  >>> i'x={x'
+  >>> t'x={x'
     File "<stdin>", line 1
-  SyntaxError: missing '}' in interpolation expression
+  SyntaxError: missing '}' in template literal expression
 
 Invalid expressions::
 
-  >>> i'x={!x}'
+  >>> t'x={!x}'
     File "<fstring>", line 1
       !x
       ^
   SyntaxError: invalid syntax
 
 Run time errors occur when evaluating the expressions inside a
-template string before creating the interpolation template object. See :pep:`498`
+template string before creating the template literal object. See :pep:`498`
 for some examples.
 
 Different renderers may also impose additional runtime
 constraints on acceptable interpolated expressions and other formatting
 details, which will be reported as runtime exceptions.
 
 
+Renderer for shell escaping added to shlex
+==========================================
+As a reference implementation, a renderer for safe POSIX shell escaping can be added to the ``shlex``
+module. This renderer would be called ``sh`` and would be equivalent of calling ``shlex.quote`` on
+each field value in the template literal.:
+
+  os.system(shlex.sh(t'cat {myfile}'))
+
+would have the same behavior as::
+
+  os.system('cat ' + shlex.quote(myfile)))
+
+The implementation would be::
+
+  def sh(template: TemplateLiteral):
+      return template.render(render_field=quote)
+
+
+Changes to subprocess module
+============================
+
+With the additional renderer in the shlex module, and the addition of template literals,
+the subprocess module can be changed to handle accepting template literal
+as an additional input type to ``Popen``, as it already accepts a sequence, or a string,
+with different behavior for each.
+With the addition of template literals, ``subprocess.Popen`` (and in return, all its higher level functions such as ``run``)
+could accept a string in a more safe way.
+For example::
+
+  subprocess.run(t'cat {myfile}', shell=True)
+
+would automatically use the ``shlex.sh`` renderer provided in this PEP. Therefore, using shlex
+inside a t-string like so::
+
+  subprocess.run(shlex.sh(t'cat {myfile}'))
+
+would be irrelevant, as ``run`` would automatically render any template literals through ``shlex.sh``
+
+
+Alternatively, when ``subprocess.Popen`` is ran without ``shell=True``, it could still provide
+subprocess with a more ergonomic syntax. For example::
+
+  subprocess.run(t'cat {myfile} --flag {value}')
+
+would be equivalent to::
+
+  subprocess.run(['cat', shlex.quote(myfile), '--flag', shlex.quote(value)])
```

Comment:
It seems wrong. For example:

``` python
>>> myfile = 'file name'
>>> open(myfile, 'wb').close()
>>> subprocess.run(['cat', shlex.quote(myfile)])
cat: "'file name'": No such file or directory
```

IMO this line

``` python
subprocess.run(t'cat {myfile} --flag {value}')
```

should be equivalent to:

``` python
subprocess.run(shlex.split(f'cat {shlex.quote(myfile)} --flag {shlex.quote(value)}'))
```

and:

``` python
subprocess.run(['cat', myfile, '--flag', value])
```

Reply:
hmm, this example is a good one. It seems you are right. I will adjust the pep.